pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -87,6 +87,12 @@ func (ssc *defaultStatefulSetControl) UpdateStatefulSet(set *apps.StatefulSet, p
 		return err
 	}
 
+	// maintain the set's revision history limit
+	defer func() {
+		err := ssc.truncateHistory(set, pods, revisions, currentRevision, updateRevision)
+		klog.Errorf("Failed to truncate revision history for StatefulSet %s/%s, error: %v", set.Namespace, set.Name, err)
```

Comment:
I think you need an if-statement here so we only log if `truncateHistory` actually returns an error.

Reply:
Oh, thanks for catching that! DONE!